From bb2374315ee2a78fe6b0e4f6274445382d8049a3 Mon Sep 17 00:00:00 2001
From: Daniel Budris <budris@amazon.com>
Date: Fri, 23 Sep 2022 10:20:16 -0400
Subject: [PATCH] add method to skip privd tests if required

---
 src/syscall/exec_linux_test.go | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/syscall/exec_linux_test.go b/src/syscall/exec_linux_test.go
index df4ed62630..1b6b5c6c11 100644
--- a/src/syscall/exec_linux_test.go
+++ b/src/syscall/exec_linux_test.go
@@ -50,6 +50,12 @@ func skipInContainer(t *testing.T) {
 	}
 }
 
+func skipPrivilegedTests(t *testing.T) {
+	if os.Getenv("SKIP_PRIVILEGED_TESTS") == "true" {
+		t.Skip("skipping this privileged test")
+	}
+}
+
 func skipNoUserNamespaces(t *testing.T) {
 	if _, err := os.Stat("/proc/self/ns/user"); err != nil {
 		if os.IsNotExist(err) {
@@ -344,6 +350,7 @@ func TestUnshareMountNameSpaceHelper(*testing.T) {
 // Test for Issue 38471: unshare fails because systemd has forced / to be shared
 func TestUnshareMountNameSpace(t *testing.T) {
 	skipInContainer(t)
+	skipPrivilegedTests(t)
 	// Make sure we are running as root so we have permissions to use unshare
 	// and create a network namespace.
 	if os.Getuid() != 0 {
@@ -386,6 +393,7 @@ func TestUnshareMountNameSpace(t *testing.T) {
 // Test for Issue 20103: unshare fails when chroot is used
 func TestUnshareMountNameSpaceChroot(t *testing.T) {
 	skipInContainer(t)
+	skipPrivilegedTests(t)
 	// Make sure we are running as root so we have permissions to use unshare
 	// and create a network namespace.
 	if os.Getuid() != 0 {
-- 
2.30.1 (Apple Git-130)

