From 562c10d67fe06953953428223743a0c12ba76316 Mon Sep 17 00:00:00 2001
From: Kirsten Schumy <ksschumy@amazon.com>
Date: Thu, 19 Jan 2023 15:33:41 -0800
Subject: [PATCH] ddkd

---
 nodeup/pkg/model/networking/common.go | 1 -
 1 file changed, 1 deletion(-)

diff --git a/nodeup/pkg/model/networking/common.go b/nodeup/pkg/model/networking/common.go
index 0f3b170fc5..e514689151 100644
--- a/nodeup/pkg/model/networking/common.go
+++ b/nodeup/pkg/model/networking/common.go
@@ -34,7 +34,6 @@ func (b *CommonBuilder) Build(c *fi.NodeupModelBuilderContext) error {
 	assets := []string{
 		"bridge",
 		"dhcp",
-		"flannel",
 		"host-device",
 		"host-local",
 		"ipvlan",
-- 
2.39.0

